Sources/Optimizely/OptimizelyClient.swift

```diff
@@ -591,12 +589,94 @@ open class OptimizelyClient: NSObject {
-                                 variableType: typeName,
+                                 variableType: variable.type,
                                  variableValue: value)
         
         return value
     }
     
+    /// Gets all the variables for a given feature.
+    ///
+    /// - Parameters:
+    ///   - featureKey: The key for the feature flag.
+    ///   - userId: The user ID to be used for bucketing.
+    ///   - attributes: The user's attributes.
+    /// - Returns: all the variables for a given feature.
+    /// - Throws: `OptimizelyError` if feature parameter is not valid
+    public func getAllFeatureVariables(featureKey: String,
+                                       userId: String,
+                                       attributes: OptimizelyAttributes? = nil) throws -> OptimizelyJSON {
+        guard let config = self.config else { throw OptimizelyError.sdkNotReady }
+        var variableMap = [String: Any]()
+        var enabled = false
+        
+        guard let featureFlag = config.getFeatureFlag(key: featureKey) else {
+            throw OptimizelyError.featureKeyInvalid(featureKey)
+        }
+        
+        let decision = self.decisionService.getVariationForFeature(config: config,
+                                                                   featureFlag: featureFlag,
+                                                                   userId: userId,
+                                                                   attributes: attributes ?? OptimizelyAttributes())
+        if let featureEnabled = decision?.variation?.featureEnabled {
+            enabled = featureEnabled
+        }
+        
+        for (_, v) in featureFlag.variablesMap {
+            var featureValue = v.value
+            if enabled, let variable = decision?.variation?.getVariable(id: v.id) {
+                featureValue = variable.value
+            }
+            
+            var valueParsed: Any? = featureValue
+            var isValidType = true
+            
+            if let valueType = Constants.VariableValueType(rawValue: v.type) {
+                switch valueType {
+                case .string:
+                    break
+                case .integer:
+                    valueParsed = Int(featureValue)
+                    break
+                case .double:
+                    valueParsed = Double(featureValue)
+                    break
+                case .boolean:
+                    valueParsed = Bool(featureValue)
+                    break
+                case .json:
+                    valueParsed = OptimizelyJSON(payload: featureValue)?.toMap()
+                    break
+                }
+            } else {
+                logger.i(.variableTypeInvalid(v.type))
+                isValidType = false
+            }
+
+            if let value = valueParsed, isValidType {
+                variableMap[v.key] = value
+                sendDecisionNotification(decisionType: .featureVariable,
```

Comment:
We're supposed to send only one notification for getAllFeatureVaraibles. See FJSON TDD spec for the new notification type.